bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ArrayValueImpl.java

```diff
@@ -656,34 +656,38 @@ public String stringValue(BLink parent) {
                 }
                 break;
             default:
-                for (int i = 0; i < size; i++) {
-                    if (refValues[i] == null) {
-                        sj.add("null");
-                    } else {
-                        Type type = TypeChecker.getType(refValues[i]);
-                        switch (type.getTag()) {
-                            case TypeTags.STRING_TAG:
-                            case TypeTags.XML_TAG:
-                            case TypeTags.XML_ELEMENT_TAG:
-                            case TypeTags.XML_ATTRIBUTES_TAG:
-                            case TypeTags.XML_COMMENT_TAG:
-                            case TypeTags.XML_PI_TAG:
-                            case TypeTags.XMLNS_TAG:
-                            case TypeTags.XML_TEXT_TAG:
-                                sj.add(((BValue) (refValues[i])).informalStringValue(new CycleUtils
-                                        .Node(this, parent)));
-                                break;
-                            default:
-                                sj.add(StringUtils.getStringValue(refValues[i], new CycleUtils.Node(this, parent)));
-                                break;
-                        }
-                    }
-                }
+                getRefValuesString(parent, sj);
                 break;
         }
         return "[" + sj + "]";
     }
 
+    private void getRefValuesString(BLink parent, StringJoiner sj) {
```

Comment:
This is a refactoring done to reduce the complexity of the method `stringValue(BLink parent)`.